graylog2-server/src/main/java/org/graylog2/indexer/SetIndexReadOnlyJob.java

```diff
@@ -92,12 +91,7 @@ public void execute() {
         activityWriter.write(new Activity("Flushed and set <" + index + "> to read-only.", SetIndexReadOnlyJob.class));
 
         if (!indexSet.get().getConfig().indexOptimizationDisabled()) {
-            try {
-                systemJobManager.submit(optimizeIndexJobFactory.create(index, indexSet.get().getConfig().indexOptimizationMaxNumSegments()));
-            } catch (SystemJobConcurrencyException e) {
-                // The concurrency limit is very high. This should never happen.
-                log.error("Cannot optimize index <" + index + ">.", e);
-            }
+            optimizeIndexJobFactory.create(index, indexSet.get().getConfig().indexOptimizationMaxNumSegments());
```

Comment:
This way we do only instantiate the job and never call `OptimizeIndexJob#execute`, or am I wrong?

Reply:
Thanks a lot for pointing this mistake!!!